test/functional/test_framework/mempool_util.py

```diff
@@ -39,19 +42,16 @@ def fill_mempool(test_framework, node, miniwallet):
     # Generate UTXOs to flood the mempool
     # 1 to create a tx initially that will be evicted from the mempool later
     # 75 transactions each with a fee rate higher than the previous one
-    test_framework.generate(miniwallet, 1 + (num_of_batches * tx_batch_size))
+    ephermal_miniwallet = MiniWallet(node, tag_name="fill_mempool_ephermal_wallet")
+    assert ephermal_miniwallet.get_scriptPubKey() != miniwallet.get_scriptPubKey()
+    test_framework.generate(miniwallet, 1)
+    test_framework.generate(ephermal_miniwallet, num_of_batches * tx_batch_size)
 
     # Mine enough blocks so that the UTXOs are allowed to be spent
     test_framework.generate(node, COINBASE_MATURITY - 1)
 
-    # Get all UTXOs up front to ensure none of the transactions spend from each other, as that may
-    # change their effective feerate and thus the order in which they are selected for eviction.
-    confirmed_utxos = [miniwallet.get_utxo(confirmed_only=True) for _ in range(num_of_batches * tx_batch_size + 1)]
-    assert_equal(len(confirmed_utxos), num_of_batches * tx_batch_size + 1)
-
     test_framework.log.debug("Create a mempool tx that will be evicted")
-    tx_to_be_evicted_id = miniwallet.send_self_transfer(from_node=node, utxo_to_spend=confirmed_utxos[0], fee_rate=relayfee)["txid"]
-    del confirmed_utxos[0]
+    tx_to_be_evicted_id = miniwallet.send_self_transfer(from_node=node, fee_rate=relayfee)["txid"]
```

Comment:
901ffa9e6e57f7c0e0db967fba9c715139bbcf0b shouldn't this be sent from the ephemeral miniwallet?

Reply:
Also I was imagining that, with the ephemeral miniwallet, we can just delete the `miniwallet` param

Reply:
Could also continue to get all the UTXOs up front?